### ISSUE TITLE
Desk not moving

### ISSUE THREAD

**Issue by jmou19** [vniehues/homebridge-linak#7]
I am running macOS and can't get the plugin to work. I have Idasen-controller installed and can run any commands easily from the terminal. But, home bridge doesn't seem to work. I know for idasen-controller, I had to put in the UUID instead of MAC address because I am on macOS. If I am running the home bridge plugin on macOS do I also need to do UUID? Also, I'm not clear on what to put on for my idasen-controller path. Should this be a path to the exe file or the site package?

**Comment by lemisieur** [comment]
Run `which idasen-controller` in your terminal to find out the path that you need to use in the plugin.
I've also used the UUID in the plugin myself instead of the mac address and it works

**Comment by vniehues** [comment]
Hi @jmou19 did you run the command that @lemisieur provided?

Basically you need to run that command from the same terminal that homebridge uses. It will print out a path which you need to use in the config. If it does not find idasen-controller, that means its not available to homebridge and you need to fix that first

**Comment by robinmassop** [comment]
Hi @vniehues !
I have the same issue running it on MacOS. The idasen-controller trough terminal is working and the desk is moving. But I can't get it to work in Homebridge. I think it has to do something with the path.

`which idasen-controller` answer:
/usr/local/bin/idasen-controller

homebridge location:
/usr/local/lib/node_modules/homebridge/bin/homebridge

If I put the location of `which idasen-controller` in homebridge, it doesn't work. Any idea?

```
"name": "Linak Platform",
            "pollingRate": 2,
            "idasenControllerPath": "/usr/local/bin/idasen-controller",
            "desks": [
                {
                    "name": "Bureau",
                    "macAddress": "BFA7FF59-503D-4249-81DC-2A03XXXXXXXX",
                    "baseHeight": 820,
                    "movementRange": 870
                }
            ],
```

**Comment by vniehues** [comment]
Hi @robinmassop! 

You need to make sure you run the command `which idasen-controller` **inside homebridge**.

The easiest way to do that is to use the homebridge UI, open the built-in terminal in the browser and then enter the command there. It either tells you that there is no idasen-controller installed (then you need to install it in homebridge) or it tells you the path you need to use.

your config looks good, i suspect that homebridge does not have access to the installation of idasen-controller and you need to install it in the homebridge-internal terminal.

please report back if that fixed your issue :)

**Comment by robinmassop** [comment]
Hi @vniehues, thanks for the response!
Any idea how to do that from inside homebridge if there's no terminal in the UI? I read that the terminal is only for Linux and not on MacOS

**Comment by vniehues** [comment]
TLDR: No :(

I just tried it. I set up an instance of HB on my mac and installed idasen-controller (ic).
ic worked well but not within HB. I tried giving node and the hb-service full disk access, tried a symlink to the hb home directory, nothing worked. 

I don't think I can really help you.

**Comment by vniehues** [comment]
@robinmassop I'm going to close this issue with that finding since the problem is not with the plugin itself but with homebridge on mac it seems... 
If you ever find a solution please come back and tell us about it! I would love to add it to the readme.

**Comment by WcEeend** [comment]
Has anyone ever found a solution for this? I'm having the same issue with Homebridge running on my Mac.
